instance/logging.py

```diff
@@ -95,3 +95,17 @@ def emit(self, record):
             # TODO: Filter out log entries for which the user doesn't have view rights
             # TODO: More targetted events - only emit events for what the user is looking at
             publish_data('log', log_event)
+
+
+class ModelLoggerAdapter(logging.LoggerAdapter):
+    """
+    Custom LoggerAdapter for model instances.
+
+    The model instance must be included under the key "obj" when constructing the logger
+    adpater.  The adapter includes information on the associated model instance by calling
+    the format_log_message() method on the instance.
+    """
+    def process(self, msg, kwargs):
+        msg, kwargs = super().process(msg, kwargs)
+        msg = self.extra['obj'].format_log_message(msg)
+        return msg, kwargs
```

Comment:
@mtyaka I first considered doing this, but there is actually one corner case that isn't covered by the less general approach: The `AppServer` class might not annotate the log message at all if it's not owned by an instance.

I copied this corner case from the old code, but thinking about it, I don't think this can ever happen, so I can make the change you suggested.